tests/cpp_unit_tests/test_math.cpp

```diff
@@ -0,0 +1,364 @@
+#include <cmath>
+#include <random>
+
+#include <boost/math/distributions/students_t.hpp>
+#include <boost/math/special_functions/legendre.hpp>
```

Comment:
This would unfortunately add Boost as a dependency, which I'm personally not keen on. Most of the reference solutions could simply be hardcoded here?

Reply:
Agreed, it'd be best to include a script that generates the reference values using `scipy` functions.

Reply:
Well, @paulromano  @gridley  thank you so much for your carefully review, I will continue to correct this PR